src/beast/beast/asio/streambuf.h

```diff
@@ -452,64 +686,65 @@ basic_streambuf<Allocator>::prepare (size_type n) ->
             out_end_ = out_pos_ + n;
             n = 0;
         }
-        debug_check();
     }
 
-    if (n > 0)
+    if(n > 0)
     {
         assert(pos == list_.end());
         for(;;)
         {
-            auto const avail = block_size_next_;
-            auto& e = *reinterpret_cast<element*>(alloc_traits::allocate(
-                this->member(), avail + sizeof(element)));
-            alloc_traits::construct(this->member(), &e, avail);
+            auto const size = std::max(alloc_size_, n);
+            auto& e = *reinterpret_cast<element*>
+                (alloc_traits::allocate(this->member(),
+                    size + sizeof(element)));
```

Comment:
This code assumes that `Allocator::pointer` is a `T*`.  This assumption can be relaxed if we want to support "smart pointers" for allocators.  Such pointers are typically used for placing `basic_streambuf` in inter-process shared memory.